core/client/src/main/java/alluxio/client/block/BlockStoreContext.java

```diff
@@ -178,29 +140,33 @@ public BlockWorkerClient acquireWorkerClient(WorkerNetAddress address)
   }
 
   /**
-   * Obtains a worker client on the local worker in the system.
+   * Obtains a worker client on the local worker in the system. For testing only.
    *
    * @return a {@link BlockWorkerClient} to a worker in the Alluxio system or null if failed
    */
   public BlockWorkerClient acquireLocalWorkerClient() {
     initializeLocalBlockWorkerClientPool();
-    if (mLocalBlockWorkerClientPool == null) {
+    if (mLocalBlockWorkerClientPoolMap.isEmpty()) {
       return null;
     }
-    return mLocalBlockWorkerClientPool.acquire();
+    // return any local worker
+    return mLocalBlockWorkerClientPoolMap.values().iterator().next().acquire();
   }
 
   /**
-   * Obtains a non local worker client based on the hostname. Illegal argument exception is thrown
-   * if the hostname is the local hostname. Runtime exception is thrown if the client cannot be
-   * created with a connection to the hostname.
+   * Obtains a worker client for the given local worker address.
    *
-   * @param hostname the worker hostname to connect to, empty string for any worker
-   * @return a worker client with a connection to the specified hostname
+   * @param address worker address
+   *
+   * @return a {@link BlockWorkerClient} to the given worker address or null if no such worker can
+   *         be found
    */
-  private BlockWorkerClient acquireRemoteWorkerClient(String hostname) {
-    WorkerNetAddress workerAddress = getWorkerAddress(hostname);
-    return acquireRemoteWorkerClient(workerAddress);
+  public BlockWorkerClient acquireLocalWorkerClient(WorkerNetAddress address) {
+    initializeLocalBlockWorkerClientPool();
+    if (!mLocalBlockWorkerClientPoolMap.containsKey(address)) {
+      return null;
+    }
+    return mLocalBlockWorkerClientPoolMap.get(address).acquire();
```

Comment:
Although we use a concurrent data structure here, this method is still not thread safe right?